components/XMLUploadButton/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Button, Input } from "@chakra-ui/react";
+import { ChangeEvent, useRef, useState } from "react";
+import { FiUpload } from "react-icons/fi";
+import { toast } from "react-toastify";
+import React from "react";
+
+interface HTMLInputEvent extends ChangeEvent {
+  target: HTMLInputElement & EventTarget;
+}
+
+/**
+ *
+ * @return {toast}
+ */
+export function XMLUploadButton() {
+  const fileRef = useRef() as React.MutableRefObject<HTMLInputElement>;
+  const [fileName, setFileName] = useState<string | undefined>("Enviar XML");
+
+  const handleChange = (e: HTMLInputEvent) => {
+    if (!e.target.files?.length) {
+      return;
+    }
+
+    const files = e.target.files?.[0];
+
+    if (files?.name.split(".").pop() != "xml") {
```

Comment:
Sempre dar preferência para utilizar o `!==` ou `===` para garantir que uma coisa seja realmente igual a outra.
Caso contrário você pode acabar tendo alguns comportamentos bizarros.

Exemplo:
![image](https://user-images.githubusercontent.com/6539258/160515487-ff368e9d-7185-4379-9ff9-4008af2a994e.png)


```suggestion
    if (files?.name.split(".").pop() !== "xml") {
```